test/gpu/native/interop/gpuLibrary/udf2.prediff

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+CHPL=$3
+OUTFILE=$2
+CHPL_HOME=`$CHPL --print-chpl-home`
+
+
+# build with compileline
+cxx=`$CHPL_HOME/util/config/compileline --compile-c++`
+libs=`$CHPL_HOME/util/config/compileline --libraries`
+$cxx test_udf2.cpp -Ilib -Llib/ -ludf2 $libs -o test_udf2_compileline
+echo "Running test_udf2_compileline" >> $OUTFILE
+./test_udf2_compileline >> $OUTFILE
+
+# build with cmake
+mkdir cmakeBuild
+cmake -B cmakeBuild -S .
+cmake --build cmakeBuild --target test_udf2_cmake
+echo "Running test_udf2_cmake" >> $OUTFILE
+./cmakeBuild/test_udf2_cmake >> $OUTFILE
+
+# build with makefile
+make test_udf2_make
+echo "Running test_udf2_make" >> $OUTFILE
+./test_udf2_make >> $OUTFILE
+
```

Comment:
This is a prediff abuse, isn't it? Probaby `sub_test` is the right approach here, but it can definitely be more cumbersome. Could you add a README to this directory you're adding?

Reply:
This is the same kind of testing we do with other interop/library tests. Almost all of them use a prediff to build the executable and then run it in some way

Reply:
:( maybe a comment in the `noexec` file to say that this test is run in `prediff` then?